packages/cosmo-dycore/package.py

```diff
@@ -79,11 +79,11 @@ def setup_environment(self, spack_env, run_env):
             spack_env.set('MPICH_G2G_PIPELINE', '64')
             if '+cuda' in self.spec:
                 spack_env.set('MPICH_RDMA_ENABLED_CUDA', '1')
-        spack_env.set('UCX_MEMTYPE_CACHE', 'n')
+        run_env.prepend_path('UCX_MEMTYPE_CACHE', 'n')
         if '+cuda' in self.spec:
-            spack_env.set('UCX_TLS', 'rc_x,ud_x,mm,shm,cuda_copy,cuda_ipc,cma')
+            run_env.prepend_path('UCX_TLS', 'rc_x,ud_x,mm,shm,cuda_copy,cuda_ipc,cma')
         else:
-            spack_env.set('UCX_TLS', 'rc_x,ud_x,mm,shm,cma')
+            run_env.prepend_path('UCX_TLS', 'rc_x,ud_x,mm,shm,cma')
```

Comment:
`setup_environment` doesn't do anything I believe?
Neither `CMakePackage` [1] nor `PackageBase`[2] have this method.
It should be inside one of the `setup_*` methods of `PackageBase`?

[1]: https://spack.readthedocs.io/en/latest/spack.build_systems.html#spack.build_systems.cmake.CMakePackage
[2]:  https://spack.readthedocs.io/en/latest/spack.html#spack.package.PackageBase

Reply:
No you're wrong, there are three methods: 
general: `setup_environment` where you can set BOTH env is for build and run_env is for run
build: `setup_build_environment`
run: `setup_run_environment`